iceoryx2-bb/lock-free/src/spmc/unrestricted_atomic.rs

```diff
@@ -73,26 +74,142 @@ impl<T: Copy> Producer<'_, T> {
         // After writing the content of the write_cell, the content needs to be synced with the
         // reader.
         /////////////////////////
-        self.atomic.write_cell.fetch_add(1, Ordering::Release);
+        self.atomic.mgmt.write_cell.fetch_add(1, Ordering::Release);
     }
 }
 
 impl<T: Copy> Drop for Producer<'_, T> {
     fn drop(&mut self) {
-        self.atomic.has_producer.store(true, Ordering::Relaxed);
+        self.atomic.mgmt.has_producer.store(true, Ordering::Relaxed);
     }
 }
 
 unsafe impl<T: Copy> Send for Producer<'_, T> {}
 unsafe impl<T: Copy> Sync for Producer<'_, T> {}
 
+#[doc(hidden)]
+#[repr(C)]
+pub struct UnrestrictedAtomicMgmt {
+    write_cell: IoxAtomicU32,
+    has_producer: IoxAtomicBool,
+}
+
+impl Default for UnrestrictedAtomicMgmt {
+    fn default() -> Self {
+        Self {
+            write_cell: IoxAtomicU32::new(1),
+            has_producer: IoxAtomicBool::new(true),
+        }
+    }
+}
+
+impl UnrestrictedAtomicMgmt {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// # Safety
+    ///
+    ///   * store operations are only allowed when this method returns Ok
+    ///   * __internal_release_producer must be called when the UnrestrictedAtomicMgmt (used
+    ///     without UnrestrictedAtomic) is dropped
+    pub unsafe fn __internal_acquire_producer(&self) -> Result<bool, bool> {
+        self.has_producer
+            .compare_exchange(true, false, Ordering::Relaxed, Ordering::Relaxed)
+    }
+
+    /// # Safety
+    ///
+    ///   * store operations are not allowed after this method was called
+    ///   * __internal_acquire_producer must have been successfully called before
+    pub unsafe fn __internal_release_producer(&self) {
+        self.has_producer.store(true, Ordering::Relaxed);
+    }
+
+    /// # Safety
+    ///
+    ///   * __internal_acquire_producer must have been successfully called before
+    ///   * the memory position must not be modified after __internal_update_write_cell has been
+    ///     called
```

Comment:
Since the memory is not synced in line 140 should we also add, that it is undefined behavior when the memory content is read. The user shall only write to it since the memory is not synchronized.